src/dispatch/DataPlaneClient.ts

```diff
@@ -114,6 +114,11 @@ export class DataPlaneClient {
         const options = {
             method: METHOD,
             protocol: this.config.endpoint.protocol,
+            port:
+                !this.config.endpoint.port ||
+                isNaN(parseInt(this.config.endpoint.port, 10))
+                    ? undefined
```

Comment:
Is there test coverage for when the port is not a number, and becomes undefined?

Reply:
Almost all the other tests are for the use-case where port is not set and I have added one where port is set. I cannot add a case for invalid port because URL class does not allow that.

Reply:
I added couple of more tests to specifically test when port is not set and validated it against the code without my change.